enumfields/fields.py

```diff
@@ -34,7 +36,11 @@ def to_python(self, value):
         raise ValidationError('%s is not a valid value for enum %s' % (value, self.enum), code="invalid_enum_value")
 
     def get_prep_value(self, value):
-        return None if value is None else self.enum(value).value
+        if value is None:
+            return None
+        if not isinstance(value, self.enum):
+            return self.enum(value).value
+        return value.value  # Already the correct type
```

Comment:
Maybe this would still be better the other way around:
```python
if value is None:
    return None
elif isinstance(value, self.enum):
    return value.value
return self.enum(value).value
```